### PR TITLE
Migrate C++ code from starboard/atomic.h to std::atomic

### PR DESCRIPTION
**Pull request by iuriionishchenko**
Migrate C++ code from starboard/atomic.h to std::atomic

b/301268304

Change-Id: I67161b7dac0a087b2b930218d12cc0e6790db291

### CODE REVIEW

File: starboard/common/spin_lock.cc [hunk 1]
```diff
@@ -18,16 +18,17 @@
...
+  while (!atomic->compare_exchange_weak(expected,
+          kSpinLockStateAcquired, std::memory_order_acquire)) {
+    expected = kSpinLockStateReleased;
     sched_yield();
   }
 }
 
-void SpinLockRelease(SbAtomic32* atomic) {
-  SbAtomicRelease_Store(atomic, kSpinLockStateReleased);
+void SpinLockRelease(std::atomic_int* atomic) {
```
Comment (y4vor) [inline starboard/common/spin_lock.cc:30]: atomic_int32_t?

File: starboard/common/spin_lock.cc [hunk 2]
```diff
@@ -42,7 +43,7 @@ void SpinLock::Release() {
   SpinLockRelease(&atomic_);
 }
 
-ScopedSpinLock::ScopedSpinLock(SbAtomic32* atomic) : atomic_(atomic) {
+ScopedSpinLock::ScopedSpinLock(std::atomic_int* atomic) : atomic_(atomic) {
```
Comment (y4vor) [inline starboard/common/spin_lock.cc:46]: same here and the rest of the files

File: starboard/nplb/media_set_audio_write_duration_test.cc [hunk 3]
```diff
@@ -151,8 +152,9 @@ class SbMediaSetAudioWriteDurationTest
   int index_ = 0;
   int64_t total_duration_ = kDuration;
   // Guard access to |pending_decoder_status_|.
-  mutable SbAtomic32 pending_decoder_status_lock_ =
-      starboard::kSpinLockStateReleased;
+  mutable std::atomic_int pending_decoder_status_lock_ {
```
Comment (y4vor) [inline starboard/nplb/media_set_audio_write_duration_test.cc:155]: atomic_int32_t?

File: starboard/common/spin_lock.cc [hunk 4]
```diff
@@ -18,16 +18,17 @@
 
 namespace starboard {
 
-void SpinLockAcquire(SbAtomic32* atomic) {
-  while (SbAtomicAcquire_CompareAndSwap(atomic, kSpinLockStateReleased,
-                                        kSpinLockStateAcquired) ==
-         kSpinLockStateAcquired) {
+void SpinLockAcquire(std::atomic_int* atomic) {
```
Comment (y4vor) [inline starboard/common/spin_lock.cc:21]: atomic_int32_t?

File: starboard/common/spin_lock.h [hunk 5]
```diff
@@ -21,15 +21,15 @@
...
 namespace starboard {
 
-const SbAtomic32 kSpinLockStateReleased = 0;
-const SbAtomic32 kSpinLockStateAcquired = 1;
+constexpr int kSpinLockStateReleased{0};
+constexpr int kSpinLockStateAcquired{1};
 
-void SpinLockAcquire(SbAtomic32* atomic);
-void SpinLockRelease(SbAtomic32* atomic);
+void SpinLockAcquire(std::atomic_int* atomic);
```
Comment (y4vor) [inline starboard/common/spin_lock.h:31]: atomic_int32_t?

**Review by y4vor — COMMENTED** [review]
Looks Great!

Just try using atomic_int32_t instead of atomic_int.

File: starboard/common/spin_lock.cc (hunk 1 above)
Comment (iuriionishchenko) [reply starboard/common/spin_lock.cc]: atomic_int32_t is optional type and might be not available for some platforms. I used atomic<int32_t> instead.

File: starboard/common/spin_lock.cc (hunk 2 above)
Comment (iuriionishchenko) [reply starboard/common/spin_lock.cc]: atomic<int32_t> is used instead

File: starboard/nplb/media_set_audio_write_duration_test.cc (hunk 3 above)
Comment (iuriionishchenko) [reply starboard/nplb/media_set_audio_write_duration_test.cc:155]: atomic_int32_t is optional type and might be not available for some platforms. I used atomic<int32_t> instead.

File: starboard/common/spin_lock.cc (hunk 4 above)
Comment (iuriionishchenko) [reply starboard/common/spin_lock.cc]: atomic_int32_t is optional type and might be not available for some platforms. I used atomic<int32_t> instead.

File: starboard/common/spin_lock.h (hunk 5 above)
Comment (iuriionishchenko) [reply starboard/common/spin_lock.h]: atomic_int32_t is optional type and might be not available for some platforms. I used atomic<int32_t> instead.

**Review by y4vor — APPROVED** [review]

**Review by amurovanyi — APPROVED** [review]

**Pull request merged**